frontend/src/features/leases/detail/LeasePages/payment/PaymentsContainer.test.tsx

```diff
@@ -0,0 +1,178 @@
+import userEvent from '@testing-library/user-event';
+import axios from 'axios';
+import MockAdapter from 'axios-mock-adapter';
+import { Claims } from 'constants/claims';
+import { LeaseStateContext } from 'features/leases/context/LeaseContext';
+import { Formik } from 'formik';
+import { createMemoryHistory } from 'history';
+import { defaultFormLease, defaultFormLeaseTerm, defaultLease } from 'interfaces';
+import { noop } from 'lodash';
+import { mockLookups } from 'mocks/mockLookups';
+import { lookupCodesSlice } from 'store/slices/lookupCodes';
+import { fillInput, renderAsync, RenderOptions, waitFor } from 'utils/test-utils';
+
+import { IPaymentsContainerProps, PaymentsContainer } from './PaymentsContainer';
+
+jest.mock('@react-keycloak/web');
+
+const history = createMemoryHistory();
+const mockAxios = new MockAdapter(axios);
+const storeState = {
+  [lookupCodesSlice.name]: { lookupCodes: mockLookups },
+};
+const setLease = jest.fn();
+
+describe('PaymentsContainer component', () => {
+  const setup = async (
+    renderOptions: RenderOptions &
+      Partial<IPaymentsContainerProps> & {
+        initialValues?: any;
+      } = {},
+  ) => {
+    // render component under test
+    const component = await renderAsync(
```

Comment:
I saw in another file that we have `render` and `renderAsync` that both seem to do the same. renderAsync was added later. Could you please comment on the need for that? Just learning

Reply:
Like the original react-test-library render does not return a promise so not sure how/why we would make it async

Reply:
renderasync automatically handles act related errors assuming no interactions are made to the page. So essentially it is a more concise way of rendering a component that has async logic. Without renderasync a snapshot test would need to have an async await on some part of the page that only renders after all promises have resolved.